tools/dashboard/generate_dashboard.py

```diff
@@ -19,71 +21,157 @@
 from pprint import pformat
 from glob import glob
 import itertools
-import configparser
+from configparser import ConfigParser
 import pickle
 from pathlib import Path
 import requests
 import hashlib
 
-import matplotlib as mpl
+import matplotlib as mpl  # type: ignore
 
 mpl.use("Agg")  # change backend, to run without X11
-import matplotlib.pyplot as plt
-from matplotlib.ticker import AutoMinorLocator
+import matplotlib.pyplot as plt  # type: ignore
+from matplotlib.ticker import AutoMinorLocator  # type: ignore
 
 # ======================================================================================
-class GitLog(list):
-    def __init__(self):
+class GitLog:
```

Comment:
Most of those classes look like they could profit from becoming `dataclasses`: https://docs.python.org/3/library/dataclasses.html, might need the backport from here https://pypi.org/project/dataclasses/ if only Python 3.6 is available

Reply:
Oh, I didn't know about `dataclasses`. Very nice.